src/DocLinkChecker/DocLinkChecker/Helpers/MarkdownHelper.cs

```diff
@@ -220,17 +221,29 @@ private static (PipeTable table, List<MarkdownError> errors)
         private static (PipeTable table, List<MarkdownError> errors)
             ValidatePipeTableWithText(string markdownFilePath, string markdown, Table table)
         {
-            PipeTable pipeTable = new (markdownFilePath, table.Line, table.Column);
+            PipeTable pipeTable = null;
+            int start = 0;
+            int len = 0;
+            string line = string.Empty;
             List<MarkdownError> errors = new ();
 
-            int start = table.Span.Start;
-            while (start > 0 && markdown.Substring(start, 1) != "\n")
+            try
             {
-                start--;
-            }
+                pipeTable = new (markdownFilePath, table.Line, table.Column);
+
+                start = Math.Max(table.Span.Start, 0);
+                while (start > 0 && markdown.Substring(start, 1) != "\n")
```

Comment:
thinking load: can you just remove replace the `\r\n` by `\n` (or `\r`) and simplify all this?

Reply:
Could be an optimization. Will leave that for now to minimize the testing that needs to take place again.

Reply:
Fair enough. Approved.